authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/factory/RestRequestFactory.java

```diff
@@ -84,18 +85,20 @@ public RestRequestDTO buildRequest(RestServicesConstants restService, Object req
 	checkHttpMethod(request, httpMethod);
 
 	if (requestBody != null) {
-	    if (!headers.getContentType().includes(MediaType.MULTIPART_FORM_DATA)) {
-		request.setRequestBody(requestBody);
-	    } else {
-		if (requestBody instanceof MultiValueMap) {
-		    request.setRequestBody(requestBody);
-		} else {
-		    throw new IDDataValidationException(
-			    IdAuthenticationErrorConstants.INVALID_INPUT_PARAMETER.getErrorCode(),
-			    String.format(IdAuthenticationErrorConstants.INVALID_INPUT_PARAMETER.getErrorMessage(),
-				    "requestBody"));
+		if(headers.getContentType()!=null){
+			if (!Objects.requireNonNull(headers.getContentType()).includes(MediaType.MULTIPART_FORM_DATA)) {

```

Comment:
`headers.getContentType()!=null` needs to be added only for the first if condition. The else condition is not related to the content type, so it should not be a new common outer if condition having the null check.